hazelcast/src/test/java/com/hazelcast/internal/monitor/impl/LocalRecordStoreStatsFailoverTest.java

```diff
@@ -0,0 +1,221 @@
+import com.hazelcast.config.EvictionPolicy;
+import com.hazelcast.config.MapConfig;
+import com.hazelcast.config.MaxSizePolicy;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.internal.metrics.MetricDescriptor;
+import com.hazelcast.internal.metrics.MetricsRegistry;
+import com.hazelcast.internal.metrics.collectors.MetricsCollector;
+import com.hazelcast.map.IMap;
+import com.hazelcast.map.LocalMapStats;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.HazelcastTestSupport;
+import com.hazelcast.test.OverridePropertyRule;
+import com.hazelcast.test.TestHazelcastInstanceFactory;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static com.hazelcast.internal.metrics.MetricDescriptorConstants.MAP_METRIC_EVICTION_COUNT;
+import static com.hazelcast.internal.metrics.MetricDescriptorConstants.MAP_METRIC_EXPIRATION_COUNT;
+import static com.hazelcast.map.impl.eviction.MapClearExpiredRecordsTask.PROP_TASK_PERIOD_SECONDS;
+import static com.hazelcast.test.Accessors.getNode;
+import static com.hazelcast.test.OverridePropertyRule.set;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class LocalRecordStoreStatsFailoverTest extends HazelcastTestSupport {
+
+    @Rule
+    public final OverridePropertyRule overrideTaskSecondsRule
+            = set(PROP_TASK_PERIOD_SECONDS, String.valueOf(1));
+
+    private static final String DEFAULT_MAP_NAME = "default";
+    private static final String EVICTABLE_MAP_NAME = "evictable-map-name";
+
+    @Override
+    protected Config getConfig() {
+        Config config = smallInstanceConfigWithoutJetAndMetrics();
+        MapConfig mapConfig = config.getMapConfig(EVICTABLE_MAP_NAME);
+        mapConfig.getEvictionConfig().setSize(1_000)
+                .setEvictionPolicy(EvictionPolicy.LRU)
+                .setMaxSizePolicy(MaxSizePolicy.PER_NODE);
+        return config;
+    }
+
+    @Test
+    public void eviction_and_expiration_counts_preserved_when_node_bouncing() {
+        Config config = getConfig();
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory();
+        HazelcastInstance instance1 = factory.newHazelcastInstance(config);
+        HazelcastInstance instance2 = factory.newHazelcastInstance(config);
+        HazelcastInstance instance3 = factory.newHazelcastInstance(config);
+
+        AtomicBoolean stop = new AtomicBoolean();
+
+        spawn(() -> {
+            while (!stop.get()) {
+                sleepSeconds(2);
+                instance3.shutdown();
```

Comment:
I would expect `instance3` is bouncing but it is never restarted. Also, could `BounceMemberRule` be helpful here?
Should this be marked as `SlowTest` before merge?

Reply:
fixed.